api/spec/src/subscription.tsp

```diff
@@ -0,0 +1,390 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/openapi3";
+import "@typespec/versioning";
+
+import "./types.tsp";
+import "./errors.tsp";
+import "./prices.tsp";
+import "./ratecards.tsp";
+import "./jsonpatch.tsp";
+import "./discounts.tsp";
+import "./entitlements/main.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.OpenAPI;
+using TypeSpec.Versioning;
+
+namespace OpenMeter;
+
+@route("/api/v1/subscriptions")
+@tag("Subscriptions")
+interface Subscriptions {
+  @get
+  @operationId("getSubscription")
+  @sharedRoute
+  get(
+    @path subscriptionId: ULID,
+    @query expand?: false,
+
+    /**
+     * The time at which the subscription should be queried. If not provided the current time is used.
+     */
+    @query at?: DateTime,
+  ): Subscription | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  @get
+  @operationId("getSubscription")
+  @sharedRoute
+  getExpanded(
+    @path subscriptionId: ULID,
+    @query expand: true,
+
+    /**
+     * The time at which the subscription should be queried. If not provided the current time is used.
+     */
+    @query at?: DateTime,
+  ): SubscriptionExpanded | OpenMeter.NotFoundError | OpenMeter.CommonErrors;
+
+  create(
+    @body body: {
+      customerId: ULID;
+      templatingPlan: {
+        planKey: Key;
+        planVersion: integer;
+      };
+      currency: CurrencyCode;
+      activeFrom: DateTime;
+
+      /**
+       * JSON Patch (RFC6902).
+       *
+       * It is a partial implementation of the JSON Patch standard:
+       * 1. It only respecting the `add` and `remove` operations. All other operations defined in RFC6902 are ignored during execution.
+       * 2. The provided value for ADD operation doesn't match the target type. It is the creation input type for the SubscriptionItem.
+       */
+      @maxItems(100)
+      overrides: SubscriptionItemOverride[];
+    },
+  ): Subscription | OpenMeter.ConflictError | OpenMeter.CommonErrors;
+
+  /**
+   * Applies the JSON Patch (RFC6902) atomically (RFC5789).
+   *
+   * It is a partial implementation of the JSON Patch standard:
+   * 1. It only respecting the `add` and `remove` operations. All other operations defined in RFC6902 are ignored during execution.
+   * 2. The provided value for ADD operation doesn't match the target type. It is the creation input type for the SubscriptionItem and SubscriptionPhase.
+   */
+  @patch
+  @operationId("editSubscription")
+  @route("/{subscriptionId}")
+  edit(
+    @header accept: "json-patch+json",
```

Comment:
Is this header required or nice to have? We learned from the ingest endpoint that custom content types are challenging for users.

Reply:
it is part of the spec